views/default/elements/forms/label.php

```diff
@@ -15,10 +16,11 @@
 }
 
 if ($required) {
-	$label .= elgg_format_element('span', [
+	$default_indicator = elgg_format_element('span', [
 		'title' => elgg_echo('field:required'),
 		'class' => 'elgg-required-indicator',
-	], "&ast;");
+			], "&ast;");
+	$label .= elgg_extract('required_indicator', $vars, $default_indicator);
```

Comment:
I'm not crazy about this pattern (render something then throw it away if not needed). In the past this has led to needless more expensive view renderings. IMO it's also more readable to put all the $vars extractions at the top.

```php
// extractions
...
// set defaults if missing values
...
// render logic
```